src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -2170,7 +2171,24 @@ QueryTreeNodePtr QueryAnalyzer::tryResolveIdentifierFromAliases(const Identifier
         auto & alias_identifier_node = it->second->as<IdentifierNode &>();
         auto identifier = alias_identifier_node.getIdentifier();
         auto lookup_result = tryResolveIdentifier(IdentifierLookup{identifier, identifier_lookup.lookup_context}, scope, identifier_resolve_settings);
-        it->second = lookup_result.resolved_identifier;
+        if (lookup_result.isResolved())
+        {
+            it->second = lookup_result.resolved_identifier;
+        }
+        else
+        {
+            alias_name_to_node_map.erase(it);
+
+            std::unordered_set<Identifier> valid_identifiers;
+            collectScopeWithParentScopesValidIdentifiersForTypoCorrection(identifier, scope, true, false, false, valid_identifiers);
+
+            auto hints = collectIdentifierTypoHints(identifier, valid_identifiers);
+            throw Exception(ErrorCodes::UNKNOWN_IDENTIFIER, "Unknown {} identifier '{}' in scope {}{}",
```

Comment:
Can we just erase?

In
```sql
SELECT min(b), x AS b FROM (SELECT max(number) FROM numbers(1));
```

we want to have error `Unknown expression identifier 'x'`, without it we would have `Unknown expression identifier 'b'`

but 
```sql
SELECT min(b), x AS b FROM (SELECT number AS x FROM numbers(1, x));
```

returns `Unknown expression identifier 'number'` now, which is not correct

Reply:
If we do not resolve identifier node, when we try to resolve some identifier from aliases, better to throw exception, because identiifer already bind to that alias.